### PR TITLE
Fix Essay Solution Parsing and Add Basic Essay Problem Response Processing

### PR DESCRIPTION
**Pull request by kaizoku**
This PR updates the method of parsing example solutions for Common Cartridge essay problems (cc.essay.v0p1) to be more specific as some Common Cartridge authoring software uses this field slightly differently.
It also adds basic response processing support for essay problems based around how Canvas generates example text for essay type problems. This fills in the ORA rubric's criterion fields with correct, incorrect, and general feedback fields to be referenced when performing assessments in the ORA review process.

**JIRA tickets**: Implements TNL-8016, BB-3863.

**Discussions**: Discussed in meetings and via email chains

**Dependencies**: None

**Merge deadline**: ASAP

**Testing instructions**:

1. Convert Common Cartridge using an essay type problem with example responses
2. Check that ORA criterion are filled from the example responses in the Common Cartridge fields

**Author notes and concerns**:

1. This only implements basic response processing for essay types based on how Canvas fills in these fields
2. If we simply want to fix the issue seen, the first commit (f768c31016539697591c860805dbc6fb7f36cf2b) will be sufficient

**Reviewers**
- [ ] @farhaanbukhsh
- [ ] @alangsto

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @kaizoku! I've created [OSPR-5658](https://openedx.atlassian.net/browse/OSPR-5658) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by alangsto** [comment]
@kaizoku thanks for putting this up! Do you have plans to add tests?

**Comment by natabene** [comment]
@kaizoku Thank you for your contribution. Please let us know once it is ready for our review.

**Comment by viadanna** [comment]
@kaizoku

I've tested this working correctly, with the converted course imported into a sandbox and criterion populated. The code also looks good.

Let me know when tests are implemented or the PR is split so I can approve it 👍

**Comment by kaizoku** [comment]
@alangsto, yes, I'm working on the tests as well. Should we perhaps split this PR between the commits?
The first commit will fix the immediate parsing issue, but it won't retain the example solution in the course that was running into this issue. That'll require the response processing added.

**Comment by alangsto** [comment]
@kaizoku I'm fine keeping them in the same PR. This issue is no longer as urgent, so I am okay waiting for the tests to be added before we merge

**Comment by kaizoku** [comment]
Thanks @viadanna. I've added to the fixtures to hit the new code from the diff and codecov is now showing 100% of diff hit.

**Review by viadanna — APPROVED** [review]
Good to go 👍

- [x] I tested this by converting the course and loading it into a sandbox.
- [x] I read through the code.

**Comment by edx-webhook** [comment]
@kaizoku 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/cc2olx&entry.867055334=https://github.com/edx/cc2olx/pull/51&entry.1484655318=https://github.com/kaizoku&entry.752974735=2021-03-08+12:30&entry.1917517419=2021-03-09+13:32&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
